From 57a65b6af1b2e8cdd34ba0c2e1fd6ce5b382353e Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:12:59 -0400
Subject: [ethernet] i40evf: don't wait so long

Message-id: <1409663591-32531-137-git-send-email-sassmann@redhat.com>
Patchwork-id: 90615
O-Subject: [RHEL7.1 PATCH 136/148] i40evf: don't wait so long
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

We really don't need to delay an entire millisecond just to get into our
critical section. A microsecond will be sufficient, thank you.

Change-ID: I2d02ece6610007d98cabcb3f42df9a774bb54e59
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit b65476cd53e074802661f4bdd9bc279b8ab65a23)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40evf_main.c b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
index ed1eb12..a53e81b 100644
--- a/drivers/net/ethernet/intel/i40evf/i40evf_main.c
+++ b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
@@ -768,7 +768,7 @@ i40evf_mac_filter *i40evf_add_filter(struct i40evf_adapter *adapter,
 
  while (test_and_set_bit(__I40EVF_IN_CRITICAL_TASK,
     &adapter->crit_section))
-  mdelay(1);
+  udelay(1);
 
  f = i40evf_find_filter(adapter, macaddr);
  if (NULL == f) {
@@ -840,7 +840,7 @@ static void i40evf_set_rx_mode(struct net_device *netdev)
 
  while (test_and_set_bit(__I40EVF_IN_CRITICAL_TASK,
     &adapter->crit_section))
-  mdelay(1);
+  udelay(1);
  /* remove filter if not in netdev list */
  list_for_each_entry_safe(f, ftmp, &adapter->mac_filter_list, list) {
   bool found = false;
-- 
1.7.1